priam/src/main/java/com/netflix/priam/config/IBackupRestoreConfig.java

```diff
@@ -113,4 +113,15 @@ default boolean enableV2Restore() {
     default String getBackupNotifyComponentIncludeList() {
         return StringUtils.EMPTY;
     }
-}
+
+    /**
+     * Returns ',' separated list of attribute names. If not empty, adds the specified attributes to
+     * MessageAttributes in the backup notifications. SNS filter policy needs keys in
+     * MessageAttributes in order to filter based on those keys.
+     *
+     * @return
+     */
+    default String getBackupNotificationAdditionalMessageAttrs() {
```

Comment:
Make this return a non-null ImmutableList(). The default should be an empty list.

Reply:
I am following the convention of other config methods which also return a String. Are you sure you want me to change the return type as List? For that, I will also have to call getList() instead of get() on IConfigSource. I would suggest leave it as is and parse the String while reading (which is what getBackupNotifyComponentIncludeList() also does).

Reply:
Returning a concatenated String to stand in as a list of Strings is not the best choice considering all callers are just going to split it anyway. If you insist on strictly adhering to convention, change it to a sensible one everywhere in a preliminary refactoring. I don't insist upon that though.